README.adoc

```diff
@@ -5727,6 +5727,18 @@ config
 
 This is simpler and more efficient.
 
+=== Object#yield_self vs Object#then [[object-yield-self-vs-object-then]]
+
+The method `Object#then` is preferred over the method `Object#yield_self`, since method name of `Object#yield_self` states the intention, not the behavior.
+[source,ruby]
+----
+# bad
+obj.yield_self { |x| x.do_something }
+
+# good
+obj.then { |x| x.do_something }
+----
```

Comment:
It may be better to place it under the following rule.
https://github.com/rubocop/ruby-style-guide#mapfindselectreduceincludesize

Reply:
Thank you for your review.
I think it’s better to move it there for sure, so I'll move it there.